src/Altinn.Apps/AppFrontend/react/shared/src/components/atoms/AltinnAttachment.tsx

```diff
@@ -62,6 +62,20 @@ function ListItemLink(props: any) {
     );
 }
 
+// Storage is always returning https:// links for attachments.
+// on localhost (without https) this is a problem, so we make links
+// to the same domain as window.location.host relative.
+// "https://domain.com/a/b" => "/a/b"
+const makeUrlRelativeIfSameDomain = (url: string) => {
```

Comment:
> makeUrlRelativeIfSameDomain is actually misleading here, right? It only makes it relative if the url is http, and not for https.
> I guess we could just make it relative in both cases? Seems better to me, and can't see any reasons that would cause a problem.
> If you have thought of something I haven't I suggest renaming to make it clear that we only make it relative for http.

I did not think of anything specific, but tried to be very careful to not break current behaviour. Solving this in js seems way safer than my first attempt in C#, and I think you are quite right that we can make it relative for the `<a` tag in all environments.


> While you are at it, maybe this method is better suited in src/studio/src/designer/frontend/shared/utils/urlHelper.ts? Some tests would also be neat 👼

Will do 👍